### DISCUSSION TITLE
Is it possible to convert a master to a worker node?

### DISCUSSION THREAD

**Discussion by binaryn3xus** [onedr0p/cluster-template#633] [Q&A] [answered]
I have a master node (node 1) that is trying its best to die in some way. Currently its a hanging NIC occasionally. I have hardware on the way.

This was my plan...

1. Convert the master to plain worker node
2. Add the new master node to replace it.
3. (maybe) Replace network interface with a USB to Ethernet for the replaced node to give it a little more life.

So would it be better to drain and remove the master then update the yaml configuration for adding the [new] master and [old] worker node back?

Related to #589 in a way.

**Comment by onedr0p** [comment, marked as answer]
I would remove the master node first (k delete node...) and then update your ansible inventory with the new ones and then run Ansible. Just make sure you backup your data first before continuing.
